ci: nightlies are failing

### What happened?

For Ubuntu, it seems we need to change the Maven version.

```
  ============================================================
  Installing Maven 3.9.10...
  ============================================================
  --2025-07-22 00:32:35--  https://dlcdn.apache.org/maven/maven-3/3.9.10/binaries/apache-maven-3.9.10-bin.tar.gz
  Resolving dlcdn.apache.org (dlcdn.apache.org)... 151.101.2.132, 2a04:4e42::644
  Connecting to dlcdn.apache.org (dlcdn.apache.org)|151.101.2.132|:443... connected.
  HTTP request sent, awaiting response... 404 Not Found
  2025-07-22 00:32:35 ERROR 404: Not Found.
```

I'm not sure what this is:

```
[ 13%] Linking C static library libnanoarrow.a
go: downloading google.golang.org/protobuf v1.36.6
cd /var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/arrow-adbc-1.8.0.XXXXX.xfNjPafW/cpp-build/vendor/nanoarrow && /private/var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/arrow-adbc-1.8.0.XXXXX.xfNjPafW/miniforge/envs/conda-source/bin/cmake -P CMakeFiles/nanoarrow.dir/cmake_clean_target.cmake
cd /var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/arrow-adbc-1.8.0.XXXXX.xfNjPafW/cpp-build/vendor/nanoarrow && /private/var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/arrow-adbc-1.8.0.XXXXX.xfNjPafW/miniforge/envs/conda-source/bin/cmake -E cmake_link_script CMakeFiles/nanoarrow.dir/link.txt --verbose=1
dyld[4704]: Library not loaded: @rpath/libLLVM.19.1.dylib
/private/var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/arrow-adbc-1.8.0.XXXXX.xfNjPafW/miniforge/envs/conda-source/bin/x86_64-apple-darwin13.4.0-ar qc libnanoarrow.a CMakeFiles/nanoarrow.dir/nanoarrow.c.o
  Referenced from: <99A81346-27DC-39F9-8A08-6198C266DC55> /private/var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/arrow-adbc-1.8.0.XXXXX.xfNjPafW/miniforge/pkgs/llvm-tools-19-19.1.7-he90a8e3_1/bin/llvm-ranlib-19
/private/var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/arrow-adbc-1.8.0.XXXXX.xfNjPafW/miniforge/envs/conda-source/bin/llvm-ranlib libnanoarrow.a
  Reason: tried: '/var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/arrow-adbc-1.8.0.XXXXX.xfNjPafW/install/lib/libLLVM.19.1.dylib' (no such file), '/libLLVM.19.1.dylib' (no such file), '/private/var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/arrow-adbc-1.8.0.XXXXX.xfNjPafW/miniforge/pkgs/llvm-tools-19-19.1.7-he90a8e3_1/bin/../lib/libLLVM.19.1.dylib' (no such file), '/private/var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/arrow-adbc-1.8.0.XXXXX.xfNjPafW/miniforge/pkgs/llvm-tools-19-19.1.7-he90a8e3_1/bin/../lib/libLLVM.19.1.dylib' (no such file), '/private/var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/arrow-adbc-1.8.0.XXXXX.xfNjPafW/miniforge/pkgs/llvm-tools-19-19.1.7-he90a8e3_1/bin/../lib/libLLVM.19.1.dylib' (no such file), '/private/var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/arrow-adbc-1.8.0.XXXXX.xfNjPafW/miniforge/pkgs/llvm-tools-19-19.1.7-he90a8e3_1/bin/../lib/libLLVM.19.1.dylib' (no such file), '/usr/local/lib/libLLVM.19.1.dylib' (no such file), '/usr/lib/libLLVM.19.1.dylib' (no such file, not in dyld cache)
Error running link command: Subprocess abortedmake[2]: *** [vendor/nanoarrow/libnanoarrow.a] Error 1
```

### Stack Trace

_No response_

### How can we reproduce the bug?

_No response_

### Environment/Setup

_No response_

--- a/ci/scripts/verify_ubuntu.sh
+++ b/ci/scripts/verify_ubuntu.sh
@@ -25,7 +25,7 @@
 set -euo pipefail
 
 : ${JDK:=21}
-: ${MAVEN:=3.9.10}
+: ${MAVEN:=3.9.11}
 
 main() {
     local -r source_dir="${1}"

